### ISSUE TITLE
Bumping build part when version has no suffix part results suffix being set to 0 (ex.: 0.1.0 -> 0.1.0-0.1))

### ISSUE THREAD

**Issue by n3tn0de** [c4urself/bump2version#114]
Here's my config:

```
[bumpversion]
current_version = 0.2.0
parse = (?P<major>\d+)\.(?P<minor>\d+)\.(?P<patch>\d+)(\-(?P<suffix>[a-z]+))?(\.(?P<build>\d+))?
serialize = 
	{major}.{minor}.{patch}-{suffix}.{build}
	{major}.{minor}.{patch}-{suffix}
	{major}.{minor}.{patch}

[bumpversion:part:suffix]
optional_value = prod
values = 
	dev
	prod

[bumpversion:file:VERSION]
```
If I try to run `bumpversion build` it will result in 0.2.0-0.1 which it should not.
It should throw error similar to the suffix part (`ValueError: The part has already the maximum value among...`)

bumpverion part | current_version | new_version | works correctly?
--- | ---| ---| ---
build | 0.1.0-dev | 0.1.0-dev.1  | ✅yes
suffix | 0.1.0-dev.1 | 0.1.0  | ✅yes
minor | 0.1.0 | 0.2.0-dev  | ✅yes
build | 0.2.0-dev | 0.2.0-dev.1  | ✅yes
suffix | 0.2.0-dev.1 | 0.2.0  | ✅yes
build | 0.2.0 | 0.2.0-0.1  | ❌ should throw error
suffix | 0.2.0-0.1 | n/a  | ❌<br>`ValueError: The part has already the maximum value among ['dev', 'prod'] and cannot be bumped.`
build | 0.2.0-0.1 | 0.2.0-0.1 | ❌ should throw error

**Comment by florisla** [comment]
Perhaps bump2version is not fully aware about the `build` part at all times?

Can you try to add this to your config and check again?

    [bumpversion:part:build]

**Comment by n3tn0de** [comment]
Just tried, same behavior, unfortunately.

**Comment by florisla** [comment]
Looking at your table again, it seems to me that this is actually standard behavior and not a bug.

Basically, you want to make `build` values larger than zero invalid in case `suffix` is `prod`, right?

Considering a part value to be 'final' is a feature that's not present at the moment.

Most people don't need this feature either I think.  The common practice is to bump `suffix` to `prod` when releasing, and bump `patch` or `minor` *immediately* after that. So the VCS heads never contain `prod` versions. `prod` solely occurs in the tags.

**Comment by n3tn0de** [comment]
Still if you accidentally bump build part on prod version instead of suffix, you break the scheme and need to roll back commit and delete the tag.

And I disagree on this
> and bump patch or minor immediately after that
You don't know what sort of changes will be added so bumping _immediately_ is against semver rules. 
Bump should be done after features and bugfixes passed PRs, QA, etc.

**Comment by florisla** [comment]
You're right, but you can't keep using  `2.0.0` in master either -- that would mean multiple commits share the same `prod` version. That also violates semver.

Bumping `tag` forces `suffix to be `dev`.  Later, at release time (or when committing impactful changes), you can still decide to bump `major` or `minor` as needed.

**Comment by n3tn0de** [comment]
Semver [generally concerns releases](https://softwareengineering.stackexchange.com/questions/388034/when-should-a-commit-not-be-version-tagged), not commits, but I see your reasoning.

Anyway I think we went off topic on this.

Bumping build part on `prod` should either result in error, `0.2.0-dev.1` or `0.2.0-dev` to comply with the scheme.

**Comment by florisla** [comment]
I understand your use case now.

The behaviour you want could be implemented as an optional feature in a `[part:name]` section.
